azure/services/virtualmachineimages/cache_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package virtualmachineimages
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/virtualmachineimages/mock_virtualmachineimages"
+)
+
+func TestCacheGet(t *testing.T) {
+	cases := map[string]struct {
+		location  string
+		publisher string
+		offer     string
+		sku       string
+		have      map[Key]compute.ListVirtualMachineImageResource
+		err       string
+		pass      bool
+	}{
+		"should find": {
+			location: "test", publisher: "foo", offer: "bar", sku: "baz",
+			have: map[Key]compute.ListVirtualMachineImageResource{
+				{location: "test", publisher: "foo", offer: "bar", sku: "baz"}: {Value: &[]compute.VirtualMachineImageResource{
+					{Name: to.StringPtr("foo")},
+				}},
+			},
+			pass: true,
+		},
+		"should not find": {
+			location: "test", publisher: "foo", offer: "bar", sku: "baz",
+			have: map[Key]compute.ListVirtualMachineImageResource{
+				{location: "test", publisher: "foo", offer: "bar", sku: "zab"}: {Value: &[]compute.VirtualMachineImageResource{
+					{Name: to.StringPtr("foo")},
+				}},
+			},
+			pass: false,
+		},
+	}
+
+	for name, tc := range cases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+			mockClient := mock_virtualmachineimages.NewMockClient(mockCtrl)
+			mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+
+			c := &Cache{
+				client: mockClient,
+				data:   tc.have,
+			}
+
+			val, err := c.Get(context.Background(), tc.location, tc.publisher, tc.offer, tc.sku)
+			if tc.err != "" {
+				if err == nil {
+					t.Fatalf("expected error %q, got nil", tc.err)
+				}
+				if err.Error() != tc.err {
+					t.Fatalf("expected error %q, got %q", tc.err, err.Error())
+				}
+			}
+
+			if tc.pass && val.Value == nil {
```

Comment:
Could you explain what the "pass" field means? Does that mean it worked without an error?